### PR TITLE
MG-888 - Update certs sdk tests

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a refactor as it refactors certs tests in the sdk.

## What does this do?

This pr:
- renames `valid` in issue cert to `validity`
- refactors certs tests

## Which issue(s) does this PR fix/relate to?

- Related Issue #888 

## Have you included tests for your changes?
Yes, the changes are to the tests.

## Did you document any new/modified feature?
Yes, I have updated the documentation for the sdk change.

### CODE REVIEW

File: pkg/sdk/go/sdk.go [hunk 1]
```diff
@@ -943,9 +943,9 @@ type SDK interface {
 	// IssueCert issues a certificate for a thing required for mTLS.
 	//
 	// example:
-	//  cert, _ := sdk.IssueCert("thingID", "valid", "token")
+	//  cert, _ := sdk.IssueCert("thingID", "validity", "token")
```
Comment (rodneyosodo) [inline pkg/sdk/go/sdk.go]: add a valid validity example

File: pkg/sdk/go/certs_test.go [hunk 2]
```diff
@@ -65,85 +73,86 @@ func TestIssueCert(t *testing.T) {
...
 			err:      errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrMissingID), http.StatusBadRequest),
-			svcerr:   errors.Wrap(certs.ErrFailedCertCreation, certs.ErrFailedCertCreation),
 		},
 	}
 
 	for _, tc := range cases {
-		svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.duration).Return(tc.cRes, tc.svcerr)
-
+		svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.duration).Return(tc.svcRes, tc.svcErr)
 		_, err := mgsdk.IssueCert(tc.thingID, tc.duration, tc.token)
```
Comment (rodneyosodo) [inline pkg/sdk/go/certs_test.go]: Check returned cert is what is expected

File: pkg/sdk/go/certs_test.go [hunk 3]
```diff
@@ -166,39 +175,38 @@ func TestViewCert(t *testing.T) {
...
 			token:  token,
-			cRes:   c,
+			svcRes: certs.Cert{},
+			svcErr: errors.Wrap(svcerr.ErrNotFound, repoerr.ErrNotFound),
 			err:    errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, svcerr.ErrNotFound), http.StatusNotFound),
-			svcerr: errors.Wrap(svcerr.ErrNotFound, repoerr.ErrNotFound),
 		},
 		{
-			desc:   "get cert with invalid token",
+			desc:   "view cert with invalid token",
```
Comment (rodneyosodo) [inline pkg/sdk/go/certs_test.go:224]:
```suggestion
			desc:   "view cert with missing token",
```

Add another case for invalid token

File: pkg/sdk/go/certs_test.go [hunk 4]
```diff
@@ -224,46 +232,43 @@ func TestViewCertByThing(t *testing.T) {
...
+			thingID: invalid,
+			token:   validToken,
+			svcRes:  certs.Page{Certs: []certs.Cert{}},
+			svcErr:  errors.Wrap(svcerr.ErrNotFound, repoerr.ErrNotFound),
 			err:     errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, repoerr.ErrNotFound), http.StatusNotFound),
-			svcerr:  errors.Wrap(svcerr.ErrNotFound, repoerr.ErrNotFound),
-			viewerr: errors.NewSDKError(svcerr.ErrViewEntity),
 		},
 		{
 			desc:    "get cert with invalid token",
```
Comment (rodneyosodo) [inline pkg/sdk/go/certs_test.go]: add another case for missing token

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/sdk/go/certs_test.go [hunk 5]
```diff
@@ -65,87 +88,93 @@ func TestIssueCert(t *testing.T) {
...
 			err:      errors.NewSDKErrorWithStatus(errors.Wrap(apiutil.ErrValidation, apiutil.ErrMissingID), http.StatusBadRequest),
-			svcerr:   errors.Wrap(certs.ErrFailedCertCreation, certs.ErrFailedCertCreation),
 		},
 	}
 
 	for _, tc := range cases {
-		svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.duration).Return(tc.cRes, tc.svcerr)
-
-		_, err := mgsdk.IssueCert(tc.thingID, tc.duration, tc.token)
+		svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.duration).Return(tc.svcRes, tc.svcErr)
```
Comment (rodneyosodo) [inline pkg/sdk/go/certs_test.go]: Refactor this to something like for all SDK tests too

```go
t.Run(tc.desc, func(t *testing.T) {
	svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.duration).Return(tc.svcRes, tc.svcErr)
	resp, err := mgsdk.IssueCert(tc.thingID, tc.duration, tc.token)
	assert.Equal(t, tc.err, err)
	if tc.err == nil {
		assert.Equal(t, sdkCert, resp)
		ok := svcCall.Parent.AssertCalled(t, "IssueCert", mock.Anything, tc.token, tc.thingID, tc.duration)
		assert.True(t, ok)
	}
	svcCall.Unset()
})
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by arvindh123 — APPROVED** [review]

**Review by WashingtonKK — APPROVED** [review]
LGTM

**Review by WashingtonKK — APPROVED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Pull request merged**